isomerpages/isomercms-frontend
# feat: introduce new version of NotFoundPage

## Problem



The current NotFound page is outdated and the design is different from what we [currently have on Figma](https://www.figma.com/file/7OtNYkc4s9BkFk05ConEbd/%E2%AD%90-V1.5?type=design&node-id=1547-66213).

This PR was also made to get familiar with the frontend codebase.

## Solution



**Breaking Changes**




**Features**:

- A new NotFound page is introduced to replace the existing one, and is updated to conform to the newer aspects of our codebase.

**Bug Fixes**:

- The "Back to dashboard" link previously did not work as intended when the user gets a NotFound error within their site, as the `siteName` parameter would not be present when the NotFound page is loaded (since it is loaded under a different route). This PR fixes this issue.

## Before & After Screenshots

**BEFORE**:


<img width="1512" alt="Screenshot 2023-08-23 at 11 36 36 AM" src="https://github.com/isomerpages/isomercms-frontend/assets/27919917/7d787869-5da8-4cda-a460-36b26e15abeb">

**AFTER**:


<img width="1512" alt="Screenshot 2023-08-23 at 11 36 55 AM" src="https://github.com/isomerpages/isomercms-frontend/assets/27919917/65798295-14a2-4614-a13e-b91ccdd9ca1c">

## Tests



    - Navigate to a random non-existent page outside of a site (i.e. `/not/found`), the "Back to dashboard" button should link you to `/sites`.
    - Navigate to a random non-existent page within a site (i.e. `/sites/zhongjun-test/notfound`), the "Back to dashboard button should link you to `/sites/zhongjun-test`.

## Deploy Notes




*None*
